tests/python/testing.py

```diff
@@ -204,7 +204,6 @@ def next(self, input_data: Callable) -> int:
             label=self.y[self.it].copy(),
             weight=self.w[self.it].copy() if self.w else None,
         )
-        gc.collect()  # clear up the copy, see if XGBoost access freed memory.
```

Comment:
is this change related? I put it there along with the "copy" function above to help prevent XGBoost from accessing freed memory.

Reply:
The repeated calls to gc are the main cause of slowness in external memory tests. Although I wonder if removing this line is causing the segfault in mgpu tests.

Reply:
That would be really interesting to understand why.